proto/remote_execution.proto

```diff
@@ -2316,4 +2316,28 @@ message VMMetadata {
 
   // The snapshot ID of the task that is currently executing.
   string snapshot_id = 3;
+
+  message SnapshotKey {
+    // Remote instance name associated with the snapshot.
+    string instance_name = 1;
+
+    // SHA256 hash of the Platform proto (exec properties) associated with the VM
+    // snapshot.
+    string platform_hash = 2;
+
+    // SHA256 Hash of the VMConfiguration of the paused snapshot.
+    string configuration_hash = 3;
+
+    // Git ref associated with the snapshot. For workflows, this represents the
+    // branch that was checked out when running the workflow.
+    string ref = 5;
+
+    // If set, this key corresponds to a specific snapshot run.
+    // If not set, this key should fetch the newest snapshot matching the other
+    // parameters.
+    string snapshot_id = 6;
+  }
```

Comment:
should this proto have the new version field?

relatedly, it seems like we could easily forget to update this proto if we update the one in firecracker.proto - wdyt about removing the one in firecracker.proto and having this be the canonical one? (Also maybe un-nest it from `VMMetadata` since it makes the generated code slightly less readable IMO)

Reply:
What do you think about this change? (WIP - linking for conceptual, not code, feedback) https://github.com/buildbuddy-io/buildbuddy/pull/6341/files

I agree with your rec to consolidate the protos